spec/takeaway_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "takeaway"
+
+describe Takeaway do
+  before :each do
+    @menu = double
+    allow(@menu).to receive(:dish).with("Pizza") { [{ name: "Pizza", price: 10 }] }
+  end
+
+  describe '#menu' do
+    it 'gets the full menu' do
+      expect(subject.menu).to eq subject.menu
+    end
+  end
+
+  describe '#order' do
+    it 'can recieve a single item order' do
```

Comment:
When writing tests, it's often better to be really explicit about the expectations, even at the cost of some repetition.  Here it would be clearer to use a literal representation of what you expect `order_dish` to return so that the reader doesn't need to look up at the before block.

That's what I think anyway!  What do you think?